torch/nn/_functions/rnn.py

```diff
@@ -21,8 +21,21 @@ def RNNTanhCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
     return hy
 
 
+def _match_hidden_with_input(input, hidden):
```

Comment:
I wonder if this is similar to `torch._C._infer_size` (which is btw used by [`variable_expandable`](https://github.com/pytorch/pytorch/blob/709fcfda8a2fb3f3da33cace584e6d5592210a9c/torch/autograd/_functions/utils.py#L31-L38) )?

Reply:
`_infer_size(x,y)` looks like it returns the size of broadcasting with x and y. Here I'm only trying to make dim 0 of input and hidden match (dim 1 is something else that doesn't need to match).

Reply:
I see. I was thinking if it is possible to cut-off the dimensions that you don't want to broadcast so that you can still use `_infer_size`, as was done in a [previous version of `matmul`](https://github.com/pytorch/pytorch/pull/1780/files#diff-61b893a141a037c275f12eba9a9b864fR185)

Reply:
But this is a completely minor comment on my side :)